solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -120,6 +134,21 @@ def resample_and_align(fx_obs, fx_series, obs_series, tz):
     obs_resampled = obs_resampled.dropna(how="any")
     obs_aligned, fx_aligned = obs_resampled.align(fx_series.dropna(how="any"),
                                                   'inner')
+    # another alignment step if reference forecast exists.
+    # here we drop points that don't exist in all 3 series.
+    # could set reference forecast to NaN where missing instead.
+    # could set to 0 instead.
```

Comment:
hmm not sure what the right thing to do here is. One problem I see with aligning all 3 is that only skill cares about the reference so you might unnecessarily through away a bunch of points

Reply:
Yes, that occurred to me too. We put this kind of question to stakeholders in a couple of workshops. There were no clear answers but a few things came up. The biggest problem with our current implementation is that a forecaster could simply refuse to submit forecasts when they know the forecast is hard. I think we'll need some user configuration for the alignment/filling logic - perhaps in report/trial specs but not at the level of the dashboard. Long story short: I am less concerned about the reference alignment at this point in time.